debezium-core/src/main/java/io/debezium/transforms/time/TimestampConverter.java

```diff
@@ -0,0 +1,785 @@
+                }
+            } else {
+                if (nestedStructField == null || nestedStructField.equals("")) {
+                    builder.field(structField.name(), OPTIONAL_TRANSLATORS.get(structField.schema().type().toString()));
+                } else {
+                    builder.field(structField.name(), structField.schema());
+                }
+            }
+        }
+        return builder;
+    }
+
+    /**
+     * Creates the new field value of the nested struct that matches with config.field_type.
+     * Avoids all the null value fields and converts the desired field value that matches to the config parameter field_type.
+     *
+     * @param field               the input nested Struct field.
+     * @param value               the input nested Struct field value.
+     * @param nestedUpdatedSchema updated field schema with target formats.
+     */
+    private Struct valueForNestedStruct(Field field, Struct value, Field nestedUpdatedSchema) {
+        Struct updatedNestedStructValue = new Struct(nestedUpdatedSchema.schema());
+        String tableName = databaseTableString(value.schema().name().toString());
+        for (Field nestedField : ((Struct) value.get(field)).schema().fields()) { //try to convert ((Struct) value.get(field)).get(nestedField) into a single field rather than computing every single time
+            final Object updatedNestedFieldValue;
+            final Object nestedFieldValue = ((Struct) value.get(field)).get(nestedField);
+            final Boolean fieldCoversionProvided = getFields(databaseTableString(value.schema().name().toString())).get(field.name()) != null ?
+                    getFields(databaseTableString(value.schema().name().toString())).get(field.name()).contains(nestedField.name()) : false;
+            if (nestedFieldValue == null || nestedFieldValue.equals("")) {
+                continue;
+            }
+
+            if (nestedField.schema().name() != null && !nestedField.schema().name().isEmpty() && TypeConverters.containsKey(nestedField.schema().name()) && (nestedFieldValue != null && !nestedFieldValue.equals("")) && !fieldCoversionProvided) {
+                updatedNestedFieldValue = convertTimestamp(nestedFieldValue, timestampTypeFromSchema(nestedField.schema()), nestedField, tableName, field.name() + "." + nestedField.name());
+            } else if (fieldCoversionProvided) {
+                updatedNestedFieldValue = convertTimestamp(nestedFieldValue, timestampTypeFromSchema(nestedField.schema()), nestedField, tableName, field.name() + "." + nestedField.name());
+            } else {
+                updatedNestedFieldValue = nestedFieldValue;
+            }
+            updatedNestedStructValue.put(nestedField.name(), updatedNestedFieldValue);
+        }
+        return updatedNestedStructValue;
+    }
+
+    private R applyWithSchema(R record) {
```

Comment:
I have to admit I find that code quite hard to grok. Why is there this differentiation of "nested" vs. "non-nested" cases? Aren't we always operating on a tree structure?